Show warning on deleting the identity that is currently logged in

edlerd opened the pull request:
## Done

- Show warning on deleting the identity that is currently logged in

Fixes WD-20336

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @Kxiru or @edlerd for access.
    - With a local copy of this branch, [build and run as described in the docs](../CONTRIBUTING.md#setting-up-for-development).
2. Perform the following QA steps:
    - log in with oidc or a new tls identity as per https://github.com/canonical/lxd-ui/pull/1139
    - go to permissions > identities, and delete the identity. Also try bulk delete. Both paths should contain a warning in the confirmation. For any other identity, not including the current user, the warning should not be visible.

# Screenshot
![image](https://github.com/user-attachments/assets/f39dd338-1d87-4cf6-9e9f-7125830f4a9b)

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://lxd-ui-1146.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/6333)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

Kxiru reviewed (dismissed):
I love the branch name!
Looks good to me, and it's a good thing to be able to confirm that the current user is being deleted. Any way for the "user" to be redirected to the login page after this? As currently the person is still able to view the interface, but just as though they have no permissions.

![image](https://github.com/user-attachments/assets/4ea64ffc-d7ac-475c-ae1f-95f6f74d9f90)

edlerd commented:
> Any way for the "user" to be redirected to the login page after this? As currently the person is still able to view the interface, but just as though they have no permissions.

That is a good point, we should invalidate the cache after this to revalidate the users auth. Will add this to this PR.

Kxiru reviewed (commented):
For your self-delete PR, this is the page that I get after deleting. This is after fetching/pulling. Just in case, I deleted the branch and checked it out again and repeated the process, but I'm still getting the same outcome. The user isn't actually logged out.

![image](https://github.com/user-attachments/assets/b09777f2-92a2-4fee-ae6f-e6edae4b1214)

edlerd commented:
> For your self-delete PR, this is the page that I get after deleting. This is after fetching/pulling. Just in case, I deleted the branch and checked it out again and repeated the process, but I'm still getting the same outcome. The user isn't actually logged out.

Good catch!

With OIDC users, they get immediately recreated after deletion. Because those type of users can self-register. So the user gets deleted, and the cache invalidated. On the next api request, the user is authenticated as OIDC user and lxd recreates the user. I added some logic to logout oidc if the current user is using oidc and removes themselves to avoid the re-creation.

Kxiru reviewed (approved):
- Single delete modal:
![image](https://github.com/user-attachments/assets/daebba33-10d4-4998-9445-1f7827584d76)
- Bulk delete modal:
![image](https://github.com/user-attachments/assets/af01d131-41b4-43f6-bce3-9ff29de4c55c)

- After delete:
![image](https://github.com/user-attachments/assets/5ebd7a21-86a7-4e2f-b6cd-db13eefd9d86)

LGTM! Though upon clicking "Login with SSO" again on this page, I get redirected back into the UI as the same user as before. However I am unsure as to whether this is a local caching issue or a larger bug/error. Surely a deleted user should no longer have working credentials even if this is cached. Refreshing the page does not fix the problem. Screenshot below (Just normal view of OIDC user with restricted privilieges, attempting to access a project).

![image](https://github.com/user-attachments/assets/65a4391a-a354-4754-96cd-9bedf758aaca)

edlerd commented:
> [...]upon clicking "Login with SSO" again on this page, I get redirected back into the UI as the same user as before. However I am unsure as to whether this is a local caching issue or a larger bug/error. Surely a deleted user should no longer have working credentials even if this is cached. Refreshing the page does not fix the problem. Screenshot below (Just normal view of OIDC user with restricted privilieges, attempting to access a project).

@Kxiru that is due to the user being recreated on login. OIDC users will register themselves after successful login. This is per design and should be ok.
